frontend/test/pytest/test_callback.py

```diff
@@ -395,5 +395,46 @@ def f(x):
     assert np.allclose(np.sin(1.0 / 2.0), f(1.0 / 2.0))
 
 
+@pytest.mark.parametrize(
+    "arg",
+    [0.1, jnp.array(0.1), jnp.array([0.1]), jnp.array([0.1, 0.2]), jnp.array([[1, 2], [3, 4]])],
+)
+def test_accelerate_device(arg):
+    """Test with device parameter"""
+
+    @accelerate(dev=jax.devices()[0])
+    def identity(x):
+        return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+        return identity(x)
+
+    assert np.allclose(qjitted_fn(arg), arg)
+
+
+@pytest.mark.parametrize(
+    "arg",
+    [0.1, jnp.array(0.1), jnp.array([0.1]), jnp.array([0.1, 0.2]), jnp.array([[1, 2], [3, 4]])],
+)
+def test_accelerate_no_device(arg):
+    """Test with no device parameter"""
+
+    # Notice that this identity is named identity2.
+    # It looks like JAX does not like pytest running tests in parallel with
+    # function names being similar.
+    # If changed to identity JAX will fail.
+
+    @accelerate
+    def identity2(x):
+        return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+        return identity2(x)
+
+    assert np.allclose(qjitted_fn(arg), arg)
+
+
```

Comment:
it would be nice to add some tests for:

- defining the `catalyst.accelerate` function _inside_ the QJIT.
- a test with autograph enabled (just to ensure everything continues to work in the future)
- tests for any exceptions we add
- tests where the user manually JITs
- tests where the user manually JITs _and_ specifies devices

Reply:
I have added:

* defining the catalyst.accelerate function inside the QJIT.
* a test with autograph enabled (just to ensure everything continues to work in the future)
* tests where the user manually JITs

Since the device specified as a qjit argument is deprecated, I am unsure what you mean by the last point.

Reply:
thanks! I think it's fine to ignore my final bullet point for now